app/src/main/kotlin/com/wire/android/ui/connection/ConnectionActionButton.kt

```diff
@@ -0,0 +1,199 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ *
+ */
+
+package com.wire.android.ui.connection
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalInspectionMode
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.core.os.bundleOf
+import com.sebaslogen.resaca.hilt.hiltViewModelScoped
+import com.wire.android.R
+import com.wire.android.model.ClickBlockParams
+import com.wire.android.navigation.EXTRA_CONNECTION_STATE
+import com.wire.android.navigation.EXTRA_USER_ID
+import com.wire.android.navigation.EXTRA_USER_NAME
+import com.wire.android.ui.common.button.WireButtonState
+import com.wire.android.ui.common.button.WireSecondaryButton
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.button.WirePrimaryButton
+import com.wire.kalium.logic.data.user.ConnectionState
+import com.wire.kalium.logic.data.user.UserId
+
+@Composable
+fun ConnectionActionButton(
+    userId: UserId,
+    userName: String,
+    connectionStatus: ConnectionState
+) {
+    val viewModel: ConnectionActionButtonBaseViewModel = if (LocalInspectionMode.current) {
+        ConnectionActionButtonPreviewModel
+    } else {
+        hiltViewModelScoped<ConnectionActionButtonViewModel>(
+            key = "${ConnectionActionButtonViewModel.MY_ARGS_KEY}$userId",
+            defaultArguments = bundleOf(
+                EXTRA_USER_ID to userId.toString(),
+                EXTRA_USER_NAME to userName,
+                EXTRA_CONNECTION_STATE to connectionStatus.toString()
+            )
```

Comment:
Did you check when the instance of ConnectionActionButtonViewModel gets cleared ? Like  we enter the screen having this ConnectionActionButton, we navigate back from the screen does this at this time gets cleared ? What if we scroll and more items do appear having this ConnectionActionButton, will they just spam more and more ViewModels ? Will they ever be cleared ?

Reply:
According to documentation: 

> Once you use the rememberScoped or viewModelScoped functions, the same object will be restored as long as the Composable is part of the composition, even if it temporarily leaves composition on configuration change (e.g. screen rotation, change to dark mode, etc.) or while being in the backstack.
> 
> For ViewModels, in addition to being forgotten when they're really not needed anymore, their coroutineScope will also be automatically canceled because ViewModel's onCleared method will be automatically called by this library.
> 
> 💡 Optional key: a key can be provided to the call, rememberScoped(key) { ... } or viewModelScoped(key) { ... }. This makes possible to forget an old object when there is new input data during a recomposition (e.g. a new input id for your ViewModel).
> 
> ⚠️ Note that ViewModels remembered with viewModelScoped should not be created using any of the Compose viewModel() or ViewModelProviders factories, otherwise they will be retained in the scope of the screen regardless of viewModelScoped. Also, if a ViewModel is remembered with rememberScoped its clean-up method won't be called, that's the reason to use viewModelScoped instead.

So when you go back it will be cleared but it will not be cleared when you will have it on back stack

Reply:
I wonder what they mean with temporarily leaves composition, if we scroll through the list will the connection request be ongoing for that specific view item that requested it ? I think it will it would be weird if it was otherwise